p2p/client.go

```diff
@@ -249,7 +268,12 @@ func (c *Client) listen(ctx context.Context) (host.Host, error) {
 		return nil, err
 	}
 
-	host, err := libp2p.New(libp2p.ListenAddrs(maddr), libp2p.Identity(c.privKey))
+	var host host.Host
+	if c.gater == nil {
+		host, err = libp2p.New(libp2p.ListenAddrs(maddr), libp2p.Identity(c.privKey))
+	} else {
+		host, err = libp2p.New(libp2p.ListenAddrs(maddr), libp2p.Identity(c.privKey), libp2p.ConnectionGater(c.gater))
+	}
```

Comment:
Such `if-else` statement is not "scalable" when new options are introduced, and it's also not very readable, even now.

```suggestion
	opts := []libp2p.Option{libp2p.ListenAddrs(maddr), libp2p.Identity(c.privKey)}
	if c.gater != nil {
		opts = append(opts, libp2p.ConnectionGater(c.gater))
	}
	host, err := libp2p.New(opts...)
```

Reply:
Thanks, I knew it was ugly but didn't know what to do